sklearn/ensemble/_base.py

```diff
@@ -131,13 +146,15 @@ def _validate_estimator(self, default=None):
 
         Sets the base_estimator_` attributes.
         """
-        if self.base_estimator is not None:
+        if self.estimator is not None:
```

Comment:
We not only need to deprecate `base_estimator` but `base_estimator_` as well.
Otherwise, we will be inconsistent.

I propose the following change to handle it in the base class:

```python
    def _validate_estimator(self, default=None):
        """Check the base estimator.

        Sets the `estimator_` attributes.
        """
        if self.estimator is not None and (
            self.base_estimator not in [None, "deprecated"]
        ):
            raise ValueError(
                "Both `estimator` and `base_estimator` were set. Only set `estimator`."
            )

        if self.estimator is not None:
            self._estimator = self.estimator
        elif self.base_estimator not in [None, "deprecated"]:
            warnings.warn(
                "`base_estimator` was renamed to `estimator` in version 1.2 and "
                "will be removed in 1.4.",
                FutureWarning,
            )
            self._estimator = self.base_estimator
        else:
            self._estimator = default

        if self._estimator is None:
            raise ValueError("`estimator` cannot be None.")

    # TODO: remove in 1.4
    # mypy error: Decorated property not supported
    @deprecated(  # type: ignore
        "Attribute `base_estimator_` was deprecated in version 1.2 and will be removed "
        "in 1.4. Use `estimator_` instead."
    )
    @property
    def base_estimator_(self):
        return self._estimator

    # TODO: remove in 1.4
    @property
    def estimator_(self):
        return self._estimator
```

This way with a temporary attribute `_estimator` is shared between `estimator_` and `base_estimator_` during the deprecation period.

Reply:
Ah Ok, sorry I didn't get that I should have also deprecated the attribute!

Reply:
This should be done now :)